vignettes/datatable-programming.Rmd

```diff
@@ -456,6 +456,50 @@ DT[, eval(cl)]
 DT[, cl, env = list(cl = cl)]
 ```
 
+## Error handling with classed conditions
+
+`data.table` provides specific condition classes for common operations, making it easier to handle errors programmatically. This is particularly useful when writing robust code or packages that use `data.table`. Relying on the exact text of error messages is fragile (it is not uncommon to change the wording slightly, or for the user's session not to be in English); prefer using the signal class where possible. Please also request new signal classes if they would benefit your use case.
+
+### Available error classes
+
+`data.table` now provides four specific error classes:
+
+- `dt_missing_column_error`: When referencing columns that don't exist
+- `dt_invalid_input_error`: When providing invalid input types or empty required arguments  
+- `dt_unsortable_type_error`: When trying to sort/key unsupported types
+- `dt_join_type_mismatch_error`: When column types are incompatible in joins/set operations
+
+### Usage examples
+
+```{r error_handling, error=TRUE}
+DT <- data.table(a = 1:3, b = 4:6)
+
+# Handle missing column errors specifically
+tryCatch({
+  setkey(DT, "nonexistent_col")
+}, dt_missing_column_error = function(e) {
+  cat("Missing column detected:", conditionMessage(e), "\n")
```

Comment:
nit: this will include a wasted space between the end of `conditionMessage(e)` and `\n`, so I always try and use `sprintf("Missing column detected: %s\n", conditionMesage(e))` or otherwise to use `sep=""`: `cat("Missing column detected: ", conditionMessage(e), "\n", sep="")`.